src/main/java/org/openrewrite/java/testing/mockito/MockitoJUnitToMockitoExtension.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2020 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.testing.mockito;
+
+import lombok.SneakyThrows;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Parser;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.format.AutoFormatVisitor;
+import org.openrewrite.java.search.FindFields;
+import org.openrewrite.java.testing.junit5.UpdateTestAnnotation;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.Statement;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.text.RuleBasedCollator;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Replaces JUnit 4 MockitoJUnit rules with JUnit MockitoExtension.
+ *
+ * Supported MockitoJUnit methods:
+ *  #rule()
+ *  #testRule()
+ *
+ * Does not currently support @Incubating MockitoJUnit.collector().
+ *
+ * @implNote collector() is designed to aggregate multiple verifications into a single output.
+ * Refactoring the method may be fairly complex and would likely benefit from being a separate recipe.
+ */
+public class MockitoJUnitToMockitoExtension extends Recipe {
```

Comment:
You need to add a line to src/main/resources/META-INF/rewrite/junit.yml to include this recipe into the JUnit4to5Migration recipe